src/main/java/org/folio/service/assignedusers/AssignedUsersServiceImpl.java

```diff
@@ -40,12 +48,53 @@
   private Converter<DbAssignedUser, AssignedUser> fromDbConverter;
   @Autowired
   private Converter<AssignedUser, User> userConverter;
+  @Autowired
+  private UsersLookUpService usersLookUpService;
 
+  @SneakyThrows
   @Override
   public CompletableFuture<AssignedUserCollection> findByCredentialsId(String credentialsId,
                                                                        Map<String, String> okapiHeaders) {
-    return assignedUserRepository.findByCredentialsId(toUUID(credentialsId), tenantId(okapiHeaders))
+    var dbAssignedUsers = assignedUserRepository.findByCredentialsId(toUUID(credentialsId), tenantId(okapiHeaders));
+
+    var ids = dbAssignedUsers
+      .thenApply(dbAssignedUsers1 -> dbAssignedUsers1.stream()
+        .map(DbAssignedUser::getId)
+        .collect(Collectors.toList()));
+
+    var users = new ArrayList<CompletableFuture<User>>();
+
+    ids.thenApply(idList -> idList.stream().map(id -> {
+      okapiHeaders.put(XOkapiHeaders.USER_ID, String.valueOf(id));
+      return users.add(usersLookUpService.lookUpUser(new OkapiParams(okapiHeaders)));
+    }));
+
+    var groupIds = users.stream()
+      .map(user -> user.thenApply(User::getPatronGroup)).collect(Collectors.toList());
+
+    var groups = new CompletableFuture<ArrayList<Group>>();
+
+    groupIds.forEach(groupId -> groupId.thenApply(id -> {
+      okapiHeaders.put(XOkapiHeaders.USER_ID, id);
+      return groups.thenApply(g -> {
+        try {
+          return g.add(usersLookUpService.lookUpGroup(new OkapiParams(okapiHeaders)).get());
+        } catch (InterruptedException | ExecutionException e) {
+          e.printStackTrace();
+          return null;
+        }
+      });
+    }));
+
+    var assignedUserCollection = dbAssignedUsers
       .thenApply(collectionConverter::convert);
+    var c = new AtomicInteger();
+    assignedUserCollection.thenCombine(groups,
+      (assignedUserCollection1, groupsList) -> assignedUserCollection1.getData()
+        .stream().peek(assignedUser ->
+          assignedUser.getAttributes().setPatronGroup(groupsList.get(c.incrementAndGet()).getGroup())));
```

Comment:
You can't be sure that the groups' collection has the same side or in the same order. Use group id to identify which group name to use.